skimage/feature/texture.py

```diff
@@ -307,6 +309,12 @@ def local_binary_pattern(image, P, R, method='default'):
            http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.214.6851,
            2004.
     """
+    match_datatype = re.search('int', str(image.dtype))
```

Comment:
@soupault thanks for the input! I just have some doubts regarding the change.

I tried running this code for an image from skimage data of dtype uint8, and got the following result:
```
In [30]: from skimage import data
         image = data.load('brick.png')

In [31]: image.dtype
Out[31]: dtype('uint8')

In [32]: image.dtype.kind
Out[32]: 'u'
```
This image has been used in the example for LBP [here](http://scikit-image.org/docs/dev/auto_examples/features_detection/plot_local_binary_pattern.html).
Since there are only two outcomes for `image.dtype.kind` for integer numpy arrays i.e. `i` if signed integer and `u` if unsigned integer, should I directly check these characters (not use re)?